### ISSUE TITLE
Add dumb mode support

### ISSUE THREAD

**Issue by arichiardi** [TypeStrong/ts-node#328]
Hello folks!

I am trying to call `ts-node` from inside an Emacs buffer and I see the color escape characters in there.
This problem already happened to me, in `lumo` [specifically](https://github.com/anmonteiro/lumo/issues/99).

I had to enable "dumb" mode in there, so now readline support whatsoever, was wondering whether it could be done here as well.

Thanks and good job here!

**Comment by blakeembrey** [comment]
I'm not sure I understand the issue. Do you just want to disable colours? The only time, I believe, there's colour used is for the error title. You should be able to specify `--no-color` to disable that (https://www.npmjs.com/package/chalk#chalksupportscolor), but let me know if that doesn't work.

**Comment by arichiardi** [comment]
Yes I thought there was full `readline` support but I am trying `--no-color` now.

**Comment by arichiardi** [comment]
Unfortunately it is not a problem of colors:
![2017-05-11-162403_827x660_scrot](https://cloud.githubusercontent.com/assets/1724788/25975960/5bd722b6-3666-11e7-8e24-517f7f85bdf9.png)

**Comment by arichiardi** [comment]
I think I can workaround with `NODE_NO_READLINE=1`.

**Comment by blakeembrey** [comment]
So the issue is related to the REPL specifically? I'm not sure what I can do there, it's all standard node REPL functionality - maybe someone has asked on their issues. Do you have the same issue running `node`?

**Comment by arichiardi** [comment]
Not really an expert on this, but [this](https://github.com/anmonteiro/lumo/blob/09bc54061b4bf178eb97f4c1382baf2f30cf31b1/src/js/repl.js#L259) is how it is solved in `lumo`.

**Comment by blakeembrey** [comment]
Since that's just an internal option to `readline`, I'm not sure how to interpret it. It looks like they overrode the line printing? I'm using REPL, so it's not exactly the same, but possibly I could change something (here: https://nodejs.org/api/repl.html#repl_customizing_repl_output) if I knew what the issue is. Can you share what exactly is broken? Is it the prompt, after you hit enter and it prints stuff out, everything?

**Comment by arichiardi** [comment]
You can see the print screen above for the error. That is what I see when I start it from within Emacs.
Apparently that is what he did, I am trying to understand as well 😀

**Comment by cspotcode** [comment]
Took a quick look and did some research.  tl;dr if anything, we can add support for node's `NODE_NO_READLINE` env var, to more closely mimic the built-in REPL.  Otherwise I'm not sure there's anything we should do here.

NodeJS internally does this when creating its REPL:
https://github.com/nodejs/node/blob/master/lib/internal/repl.js#L28-L30
https://github.com/nodejs/node/blob/master/lib/internal/repl.js#L50-L51

We do this: (slightly different)
https://github.com/TypeStrong/ts-node/blob/master/src/bin.ts#L345

https://www.gnu.org/software/emacs/manual/html_node/emacs/Terminal-emulator.html
Is it possible @arichiardi was not using term mode?  Or why is emacs giving a TTY if it can't handle ANSI sequences?

We could theoretically support node's `NODE_NO_READLINE` flag by copy-pasting what node does internally.  But I'm not sure that's the right solution for this particular issue.
https://nodejs.org/api/repl.html#repl_using_the_node_js_repl_with_advanced_line_editors

`^[[1G^[[0J` is the sequence to move the cursor to the beginning of the line and erase the entire line.  (not super-helpful, but I googled it, so here you go)
